proxy/src/main/java/org/apache/rocketmq/proxy/service/route/MessageQueueSelector.java

```diff
@@ -44,8 +47,9 @@ public class MessageQueueSelector {
     private final Map<String, AddressableMessageQueue> brokerNameQueueMap = new ConcurrentHashMap<>();
     private final AtomicInteger queueIndex;
     private final AtomicInteger brokerIndex;
+    private TopicRouteService topicRouteService;
 
-    public MessageQueueSelector(TopicRouteWrapper topicRouteWrapper, boolean read) {
+    public MessageQueueSelector(TopicRouteWrapper topicRouteWrapper, TopicRouteService topicRouteService, boolean read) {
```

Comment:
Why not pass MqFaultStrategy instead of TopicRouteService as TopicRouteService is not actually used in this class.